crates/control/src/behavior/intercept_ball.rs

```diff
@@ -18,25 +18,31 @@ pub fn execute(
     maximum_step_size: Step,
 ) -> Option<MotionCommand> {
     if let Some(
-        GameControllerState {
+        FilteredGameControllerState {
             game_phase: GamePhase::PenaltyShootout { .. },
             ..
         }
-        | GameControllerState {
+        | FilteredGameControllerState {
             sub_state: Some(SubState::PenaltyKick),
             ..
         },
-    ) = world_state.game_controller_state
+    ) = world_state.filtered_game_controller_state
     {
         return None;
     }
     match (
-        world_state.filtered_game_state,
+        world_state
+            .filtered_game_controller_state
+            .map(|filtered_game_controller_state| filtered_game_controller_state.game_state),
```

Comment:
please extract that into a variable before the `match`

Reply:
As discussed postponed.

Reply:
Done